fennel/datasets/datasets.py

```diff
@@ -226,8 +226,46 @@ def __add__(self, other):
     def rename(self, columns: Dict[str, str]) -> _Node:
         return Rename(self, columns)
 
-    def drop(self, columns: List[str]) -> _Node:
-        return Drop(self, columns)
+    def __drop(self, columns: List[str], name="drop") -> _Node:
+        return Drop(self, columns, name=name)
+
+    @classmethod
+    def __get_drop_args(
+        cls, *args, columns: Optional[List[str]], name="drop"
+    ) -> List[str]:
+        if args and columns is not None:
+            raise ValueError(
+                f"can only specify either 'columns' or positional arguments to {name}, not both."
+            )
+        elif columns is not None:
+            return columns
+        elif args:
+            cols = [*args]
+            if len(args) == 1 and isinstance(args[0], list):
+                cols = args[0]
+            return cols
+        else:
+            raise ValueError(
+                f"must specify either 'columns' or positional arguments to {name}."
+            )
+
+    def drop(self, *args, columns: Optional[List[str]] = None) -> _Node:
```

Comment:
Can you reorganize the code to put the public ones up and the __ ones below them